### ISSUE TITLE
Matthews Correlation Coefficient

### ISSUE THREAD

**Issue by rushilg99** [macrosynergy/macrosynergy#2180]
Add a regression metric called `regression_mcc` in accordance with the sklearn API that should return the Matthews Correlation Coefficient (https://scikit-learn.org/1.5/modules/generated/sklearn.metrics.matthews_corrcoef.html) of the signs of predictions and targets. This can be done similarly to how regression_balanced_accuracy is implemented.

**Comment by rushilg99** [comment]
One small amendment needed. It's just the probability of significance that needs to be amended, not the correlation coefficient.
